Ugly hack that prevents server termination with xf86-video-fbdev-0.4.1
(and probably other) while returning from chvt or resume on some
hardware (e. g. zaurus).

Correct fix would require debugging of fbdev mode during LeaveVT and
EnterVT.

This patch may cause staying in incorrect or corrupted display mode
after EnterVT, but on man affected devices it does not cause any visible
problems.

Hacked code is never called on properly written drivers.

Devices affected and testers involved for future patch removal:
Sharp Zaurus (spitz/akita): Stanislav Brabec <utx@penguin.cz>

See also:
https://bugzilla.redhat.com/show_bug.cgi?id=238451

The bug (first line indicates that your device is affected):

(EE) FBDEV(0): FBIOPUT_VSCREENINFO succeeded but modified mode

Fatal server error:
EnterVT failed for screen 0

Index: xorg-server-1.7.3/hw/xfree86/fbdevhw/fbdevhw.c
===================================================================
--- xorg-server-1.7.3.orig/hw/xfree86/fbdevhw/fbdevhw.c
+++ xorg-server-1.7.3/hw/xfree86/fbdevhw/fbdevhw.c
@@ -571,7 +571,7 @@ fbdevHWSetMode(ScrnInfoPtr pScrn, Displa
 #if DEBUG
 		print_fbdev_mode("returned", &set_var);
 #endif
-		return FALSE;
+		/* return FALSE; UGLY HACK to allow return from chvt */
 	}
 
 	if (!check)
